apps/server/src/modules/authentication/strategy/jwt.strategy.ts

```diff
@@ -25,14 +30,33 @@ export class JwtStrategy extends PassportStrategy(Strategy) {
 		// check jwt is whitelisted and extend whitelist entry
 		const { accountId, jti, userId } = payload;
 		await this.jwtValidationAdapter.isWhitelisted(accountId, jti);
-		// check user exists
+
+		// check if user exists
+		let user: User;
 		try {
-			await this.userRepo.findById(userId);
+			user = await this.userRepo.findById(userId);
 		} catch (err) {
-			throw new UnauthorizedException('Unauthorized.');
+			throw new UnauthorizedException();
+		}
+
+		// check if user has roles
+		if (payload.roles.length <= 0) {
+			throw new UnauthorizedException();
 		}
 
-		// TODO: check user/account is active and has one role
+		const roles: Role[] = await this.roleRepo.findByIds(payload.roles);
```

Comment:
This is NOT authentication, this is authorisation.
As such, it should not be something that is triggered by the API layer either, but handled by the authorisation service and triggered by the uc.